pkg/scaffold/internal/filesystem/mock_test.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```

Comment:
Do not show strange the need to implement tests for the mock tests?

Reply:
Hi @Adirio, 

> Because these are unitary tests. Unitary tests only test one thing, as oposed to end-to-end (e2e) tests, which test the whole process. 

We do not need to test and verify other places and structures. However, we do need to create/duplicate the "real" implementation to do these helpers. IHMO we should use the real "structures" and methods implemented to mock the data instead of that. 

@mengqiy @DirectXMan12 what is your input here? Do you see a need to create this mocks or should we use the "real" implementations and then, mock the data as usual? Are you ok with?

Reply:
Then it won't be an unitary test. It would be an e2e test. Because if we use `fileSystem` we would be creating the file in the disk. We are already doing that with the golden test.

And furthermore, if we use `fileSystem` we would need to create functions to read the data and check it is the same.